docs/writing-rules/rule-syntax.md

```diff
@@ -191,6 +192,55 @@ rules:
 
 The working example, on the other hand, will fire on both functions in the javascript test code.
 
+## `metavariable-comparison`
+
+The `metavariable-comparison` operator compares metavariables against a basic [Python comparison](https://docs.python.org/3/reference/expressions.html#comparisons) expression. This is useful for filtering results based on a [metavariable's](../writing-rules/pattern-syntax.md#metavariables) numeric value.
+
+Example:
+
+The `metavariable-comparison` operator is a mapping which requires the `metavariable` and `comparison` keys. It can be combined with other pattern operators:
+
+```yaml
+rules:
+  - id: superuser-port
+    patterns:
+      - pattern: set_port($ARG)
+      - metavariable-comparison:
+          metavariable: '$ARG'
+          comparison: '$ARG < 1024'
+    message: "module setting superuser port"
+    languages: [python]
+    severity: ERROR
+```
+
+This will catch code like `set_port(80)` or `set_port(443)`, but not `set_port(8080)`.
+
+The `metavariable-comparison` operator also takes optional `base: int` and `strip: bool` keys. These keys set the integer base the metavariable value should be interpreted as and remove quotes from the metavariable value, respectively.
+
+For example, `base`:
+
+```
+- pattern: set_permissions($ARG)
+- metavariable-comparison:
+    metavariable: '$ARG'
+    comparison: '$ARG > 0o600'
+    base: 8
+```
+
+This will interpret metavariable values found in code as octal, so `0700` will be detected, but `0500` will not.
+
+For example, `strip`:
+
+```
+- pattern: to_integer($ARG)
+- metavariable-comparison:
+    metavariable: '$ARG'
+    comparison: '$ARG > 2147483647'
+    strip: true
+```
+
+This will remove quotes (`'`, `"`, and `` ` ``) from both ends of the metavariable content. So `"2147483648"` will be detected but `"2147483646"` will not. This is useful when you expect strings to contain integer or float data.
```

Comment:
I see the numerical difference between `"2147483648"` and `"2147483646"`,  but the quotes are still included in both - am I missing something? It’s just that the comparison is against 2147483647? If that's the case, it might be more straightforward to just include one the `"2147483648"` example.

Reply:
Yeah, it's to highlight that these are strings in code, but are treated as integers in the comparison. This option isn't frequently used, but we do have a few places in `semgrep-rules` making this type of comparison. I'll roll with this for now so it's a simple copy and paste from experiments :+1: